src/Uno.UI/UI/Xaml/Controls/TimePicker/NativeTimePickerFlyout.Android.cs

```diff
@@ -26,7 +26,23 @@ internal protected override void Open()
 		AddToOpenFlyouts();
 	}
 
-	private void SaveInitialTime() => _initialTime = Time;
+	private void SaveInitialTime()
+	{
+
+		if (Time.Ticks == -1)
+		{
+			var calendar = new global::Windows.Globalization.Calendar();
+			calendar.SetToNow();
+			var currentDateTime = calendar.GetDateTime();
+			_initialTime = new TimeSpan(currentDateTime.Hour, currentDateTime.Minute, currentDateTime.Second);
+		}
+		else
+		{
+			_initialTime = Time;
+		}
+
+	}
```

Comment:
Instead of doing this in SaveTime, I would've thought this logic could be run whenever we are about to open the picker flyout. Similar to how it is done for the DatePicker native flyout: https://github.com/unoplatform/uno/blob/7b287950e524d4b722c05dba37ed6fedbd226e10/src/Uno.UI/UI/Xaml/Controls/DatePicker/NativeDatePickerFlyout.Android.cs#L43-L56

Reply:
And here for iOS: https://github.com/unoplatform/uno/blob/7b287950e524d4b722c05dba37ed6fedbd226e10/src/Uno.UI/UI/Xaml/Controls/DatePicker/NativeDatePickerFlyout.iOS.cs#L145-L157

In fact, if there are no differences between iOS and Android in terms of the current time calc, we could probably centralize this getCurrentTime logic to the [shared partial NativeTimePickerFlyout code](https://github.com/unoplatform/uno/blob/dev/ks/ios-timePicker/src/Uno.UI/UI/Xaml/Controls/TimePicker/NativeTimePickerFlyout.cs)